packages/mobile/src/invite/JoinCelo.tsx

```diff
@@ -105,15 +103,11 @@ export class JoinCelo extends React.Component<Props, State> {
     this.props.hideAlert()
   }
 
-  onPressGoToTerms = () => {
-    navigateToURI(CELO_TERMS_LINK)
-  }
-
   goToNextScreen = () => {
     const nextScreen =
       this.props.pincodeType === PincodeType.Unset
         ? Screens.PincodeEducation
```

Comment:
im not sure what the path is for PinCode.UNset and if in that flow we would also go to the terms screen, but im assuming that it would only happen after they have onboarded

Reply:
So, desired behaviour is Terms screen would be after user sets the PIN, right?

Reply:
Its before the PIN. they enter it right after the Name Country Phone Number Screen

Reply:
When app was launched for first time `pincodeType === PincodeType.Unset // true`, so right after `JoinCelo` screen user would be redirected to `PincodeEducation`, where he/she sets a Pincode and goes straight to `EnterInviteCode` screen. Thus the only way to get to `RegulatoryTerms` screen would be to reload the app, then in `navigateToProperScreen` saga we would catch it earlier than `PincodeType.Unset` condition. 

Probably, I'm missing something here - let me know. But if my understanding is correct I would suggest to add another condition to this "if" to check `acceptedTerms === true` and only then navigate to  `PincodeEducation ` screen. This, though, would require duplicating this "if" in `RegulatoryTerms` as well.

Reply:
oh! this is very helpful. Thanks man. I mistook Unset to mean we were unsetting it.

Reply:
I think i got it now

Reply:
logic like this
```
    if (!this.props.acceptedTerms) {
      navigate(Screens.RegulatoryTerms)
    } else if (this.props.pincodeType === PincodeType.Unset) {
      navigate(Screens.PincodeEducation)
    } else {
      navigate(Screens.EnterInviteCode)
    }

```

Reply:
although im a bit confused why after pincode education the following screen is InviteCode not PineCodeEnter. but that seems to be how it has been